src/main/java/me/deadlight/ezchestshop/Listeners/PlayerTransactionListener.java

```diff
@@ -89,7 +93,8 @@ private void log(PlayerTransactEvent event) {
         }
 
         data.set(new NamespacedKey(EzChestShop.getPlugin(), "trans"), PersistentDataType.STRING, finalString.toString());
-        event.getChest().update();
+        state.update();
```

Comment:
We have to try this one out later because I remember there was a dupe bug with this updating part...

Reply:
With the new or the old system? I just tried copying and adapting your existing system.